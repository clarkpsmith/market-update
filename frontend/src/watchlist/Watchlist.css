.Watchlist {
  color: white;
  background-color: black;
  height: 100vh;
}
.Watchlist-title {
  background-color: black;
  color: rgb(200, 210, 236);
  padding: 5px;
  margin-bottom: 0;
  border-bottom: 1px solid grey;
}

.Chase-loading {
  display: flex;
  justify-content: center;
}

.Watchlist-stocks {
  border-bottom: 1px solid grey;
}

.Watchlist-empty {
  padding-top: 5px;
  padding-bottom: 5px;
  background-color: black;
}
