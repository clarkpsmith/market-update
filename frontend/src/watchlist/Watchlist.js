import React, { useEffect, useState } from "react";
import Item from "./Item";
import "./Watchlist.css";
import YahooFinanceApi from "../api/YahooFinanceApi";
import SearchBlock from "../searchBlock/SearchBlock";
import { useSelector, useDispatch } from "react-redux";
import ChaseLoading from "../chaseloading/ChaseLoading";

const Watchlist = () => {
  const [loading, setLoading] = useState(true);
  const dispatch = useDispatch();
  const watchlist = useSelector((store) => store.currentUser.watchlist);
  const watchlistData = useSelector((store) => store.watchlistData);

  let watchlistString;
  if (watchlist) watchlistString = watchlist.join(",");

  useEffect(() => {
    async function getStocks(tickers) {
      const res = await YahooFinanceApi.searchTicker(tickers);

      dispatch({ type: "SET_WATCHLIST_DATA", watchlistData: res });

      setLoading(false);
    }
    if (watchlist.length > 0) {
      getStocks(watchlistString);
    } else setLoading(false);
  }, []);

  if (loading) {
    return (
      <div data-testid="Watchlist" className="Watchlist">
        <ChaseLoading />
      </div>
    );
  }

  const watchlistsArray =
    watchlistData.map((ticker) => (
      <Item key={ticker.symbol} ticker={ticker} />
    )) || "No Stocks in Watchlist";

  return (
    <div className="Watchlist">
      <div>
        <SearchBlock />
        <h3 className="Watchlist-title">Your Watchlist</h3>
      </div>
      <div className="Watchlist-stocks">
        {watchlistsArray.length > 0 ? (
          watchlistsArray
        ) : (
          <div className="Watchlist-empty">No Stocks in Watchlist</div>
        )}
      </div>
    </div>
  );
};

export default Watchlist;
