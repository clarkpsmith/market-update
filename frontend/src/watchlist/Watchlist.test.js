import React from "react";
import { render, waitForElementToBeRemoved } from "@testing-library/react";
import Watchlist from "./Watchlist";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";
import { tslaData } from "../testValues";
const mockStore = configureMockStore();

const store = mockStore({
  currentUser: {
    firstName: "test",
    lastName: "user",
    email: "testuser@gmail.com",
    username: "testuser",
    watchlist: ["tsla"],
  },
  watchlistData: [tslaData.data.quoteResponse.result[0]],
  searchTickerData: null,
});

it("renders without crashing", () => {
  render(
    <Provider store={store}>
      <Watchlist />
    </Provider>
  );
});

it("matches snapshot", async () => {
  const { asFragment, queryByTestId } = render(
    <Provider store={store}>
      <Watchlist />
    </Provider>
  );
  await waitForElementToBeRemoved(() => queryByTestId("Watchlist"));

  expect(asFragment()).toMatchSnapshot();
});
