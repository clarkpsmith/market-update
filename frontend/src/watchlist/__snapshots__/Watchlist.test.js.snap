// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Watchlist"
  >
    <div>
      <section
        class="SearchBlock"
      >
        <div
          class="SearchBlock-card"
        >
          <div
            class="SearchBlock-search"
          >
            <div
              class="SearchForm"
            >
              <form
                class="SearchForm-form form-inline"
              >
                <input
                  class="SearchForm-input"
                  id="searchTerm"
                  name="searchTerm"
                  placeholder="Enter ticker symbol..."
                  type="text"
                  value=""
                />
                <button
                  class="SearchForm-btn btn btn-secondary btn-md"
                >
                  Search
                </button>
              </form>
            </div>
          </div>
          <div
            class="results"
          />
          <div
            class="SearchBlock-no-results"
          />
        </div>
      </section>
      <h3
        class="Watchlist-title"
      >
        Your Watchlist
      </h3>
    </div>
    <div
      class="Watchlist-stocks"
    >
      <div
        class="Item"
      >
        <h4
          class="Item-title"
        >
          <span
            title="Expand"
          >
            TSLA
          </span>
          <span>
            <button
              class="Item-btn-watchlist btn btn-secondary btn-sm"
              type="button"
            >
              Add
            </button>
          </span>
          <span>
            <span
              class="Stockinfo-price"
              title="Current stock price, click to refresh"
            >
              636.29
            </span>
          </span>
        </h4>
        <div
          class="Item-fullname"
        >
          Tesla, Inc. 
          <span>
            <button
              class="Stockinfo-daily-change Item-btn-change"
              style="background-color: red;"
            >
              -5.37 %
            </button>
          </span>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
